dcos/subcommand.py

```diff
@@ -231,19 +280,32 @@ def _install_env(pkg, options):
 
     pkg_dir = _package_dir(pkg.name())
 
-    install_operation = pkg.command_json(options)
-
-    env_dir = os.path.join(pkg_dir,
-                           constants.DCOS_SUBCOMMAND_VIRTUALENV_SUBDIR)
+    resources = pkg.resource_json()
+    binary_cli = _get_cli_binary(resources)
+    env_dir = os.path.join(pkg_dir)
 
-    if 'pip' in install_operation:
-        _install_with_pip(
+    if binary_cli:
+        _install_with_binary(
             pkg.name(),
             env_dir,
-            install_operation['pip'])
+            binary_cli)
     else:
-        raise DCOSException("Installation methods '{}' not supported".format(
-            install_operation.keys()))
+        install_operation = pkg.command_json(options)
```

Comment:
Thanks!

On Wed, Apr 27, 2016 at 12:39 PM tamarrow <notifications@github.com> wrote:

> In dcos/subcommand.py
> <https://github.com/dcos/dcos-cli/pull/586#discussion_r61322519>:
>
> >      else:
> > -        raise DCOSException("Installation methods '{}' not supported".format(
> > -            install_operation.keys()))
> > +        install_operation = pkg.command_json(options)
>
> I'll fix this in another pr. Created #593
> <https://github.com/dcos/dcos-cli/issues/593>
>
> —
> You are receiving this because you commented.
>
>
> Reply to this email directly or view it on GitHub
> <https://github.com/dcos/dcos-cli/pull/586/files/b49c3487208434d7e02ce57509db71ad5c624347#r61322519>
>
-- 
José Armando García Sancio
Mesosphere